### ISSUE TITLE
next/script eslint-plugin-next Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by gapon2401** [vercel/next.js#32178]
### What version of Next.js are you using?

12.0.7

### What version of Node.js are you using?

14.17.5

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

next start

### Describe the Bug

When I'm using spread operator for `<Script>` tag, I got an error:
```
    at \node_modules\@next\eslint-plugin-next\lib\rules\inline-script-id.js:34:43
    at Array.some (<anonymous>)
    at JSXElement (\node_modules\@next\eslint-plugin-next\lib\rules\inline-script-id.js:33:22)
    at \node_modules\eslint\lib\linter\safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (\node_modules\eslint\lib\linter\safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (\node_modules\eslint\lib\linter\node-event-generator.js:293:26)
    at NodeEventGenerator.applySelectors (\node_modules\eslint\lib\linter\node-event-generator.js:322:22)
    at NodeEventGenerator.enterNode (\node_modules\eslint\lib\linter\node-event-generator.js:336:14)
    at CodePathAnalyzer.enterNode (\node_modules\eslint\lib\linter\code-path-analysis\code-path-analyzer.js:711:23)
```

### Expected Behavior

I expected to have no eslint errors

### To Reproduce

`<Script {...{ strategy: 'beforeInteractive' }} id={'test'}   />`

**Comment by balazsorban44** [comment]
I could not reproduce, could you post your repository or a minimal reproduction?

**Comment by balazsorban44** [comment]
Issue is at this line:
https://github.com/vercel/next.js/blob/c004322d138e1949437efb0b80e0973f3a3775cf/packages/eslint-plugin-next/lib/rules/inline-script-id.js#L35

**Comment by josharsh** [comment]
@balazsorban44 are you picking this up? (If not, I would like to give a shot)

**Comment by balazsorban44** [comment]
please do!

**Comment by balazsorban44** [comment]
What did was:

1. create an example project with `yarn create next-app`
2. Open in VScode (Make sure you have the [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) extension enabled)
3. Open `pages/index.js`
4. add `import Script from "next/script`
5. add `<Script {...{ strategy: "beforeInteractive" }} id={"test"} />` in the return body

VSCode shows this in the Output:

```sh
[Error - 12:49:38 AM] TypeError: Cannot read property 'name' of undefined
Occurred while linting .../pages/index.js:9
Rule: "@next/next/inline-script-id"
    at .../node_modules/@next/eslint-plugin-next/lib/rules/inline-script-id.js:35:43
    at Array.some (<anonymous>)
    at JSXElement (.../node_modules/@next/eslint-plugin-next/lib/rules/inline-script-id.js:34:22)
    at ruleErrorHandler (.../node_modules/eslint/lib/linter/linter.js:1076:28)
    at .../node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (.../node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (.../node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (.../node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (.../node_modules/eslint/lib/linter/node-event-generator.js:340:14)
```

**Comment by josharsh** [comment]
Not able to reproduce.
Here are the specifics:
* next version: v12.0.7
* node version: v16.6.2
* OS: mac
* Running Using: yarn dev

Is there anything I am missing?

**Comment by balazsorban44** [comment]
Try running `yarn lint`?:

```sh
TypeError: Cannot read properties of undefined (reading 'name')
Occurred while linting /home/balazs/projects/next-support/32178/pages/index.js:9
Rule: "@next/next/inline-script-id"
error Command failed with exit code 1.
```

**Comment by stefanprobst** [comment]
i think changing https://github.com/vercel/next.js/blob/c004322d138e1949437efb0b80e0973f3a3775cf/packages/eslint-plugin-next/lib/rules/inline-script-id.js#L30 like this should work:

```diff
- const attributes = node.openingElement.attributes
+ const attributes = node.openingElement.attributes.filter(attribute => attribute.type === 'JSXAttribute')

```

**Comment by Xunnamius** [comment]
Might be related (if not, I'll open a new issue), but I'm receiving the same error but in a totally different place:

```shell
$ npx eslint --parser-options=project:tsconfig.lint.json src

Oops! Something went wrong! :(

ESLint: 8.4.1

TypeError: Cannot read properties of null (reading 'name')
Occurred while linting /repos/.ergodark/xunn.at/src/pages/api/[shortId].ts:9
Rule: "@next/next/no-page-custom-font"
    at ExportDefaultDeclaration (/repos/.ergodark/xunn.at/node_modules/@next/eslint-plugin-next/lib/rules/no-page-custom-font.js:46:56)
    at ruleErrorHandler (/repos/.ergodark/xunn.at/node_modules/eslint/lib/linter/linter.js:1076:28)
    at /repos/.ergodark/xunn.at/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/repos/.ergodark/xunn.at/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/repos/.ergodark/xunn.at/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/repos/.ergodark/xunn.at/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/repos/.ergodark/xunn.at/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/repos/.ergodark/xunn.at/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
    at /repos/.ergodark/xunn.at/node_modules/eslint/lib/linter/linter.js:1107:32
```

It's doubly strange because I'm mostly doing API stuff so there is no frontend code yet, let alone any code having to do with fonts. Removing `'plugin:@next/next/recommended'` from my ESLint `extends` "solves" the problem.

**Comment by balazsorban44** [comment]
@Xunnamius I believe that is already fixed #32251
